modules/search/instant-search/lib/query-string.js

```diff
@@ -220,3 +220,12 @@ export function restorePreviousHref( initialHref ) {
 		window.dispatchEvent( new Event( 'queryStringChange' ) );
 	}
 }
+
+export function clearSearchAndFilterQueries() {
+	// TODO: Improve detection of search and filter queries
```

Comment:
As mentioned in the todo, it would be nice to look for the specific URL parameters we use (like `s` and `post_tag`).

Reply:
Pushed out a change that basically accomplishes this.